File: example/android/awn_core/src/main/java/me/carda/awesome_notifications/core/managers/LocalizationManager.java
package me.carda.awesome_notifications.core.managers;

import android.content.Context;

import java.util.Locale;

import me.carda.awesome_notifications.core.databases.SQLitePrimitivesDB;
import me.carda.awesome_notifications.core.exceptions.ExceptionCode;
import me.carda.awesome_notifications.core.exceptions.ExceptionFactory;

public class LocalizationManager {
    private static final String TAG = "LocalizationManager";

    private static LocalizationManager instance;
    public static LocalizationManager getInstance(){
        if (instance == null) instance = new LocalizationManager();
        return instance;
    }

    public boolean setLocalization(Context context, String languageCode){
        try {
            if (languageCode == null) {
                Locale currentLocale = context.getResources().getConfiguration().locale;
                languageCode = currentLocale.toLanguageTag().toLowerCase();
            }
            SQLitePrimitivesDB
                    .getInstance(context)
                    .setString(
                        context,
                        "localization",
                        "languageCode",
                        languageCode.toLowerCase()
                    );
            return true;
        } catch (Exception e) {
            ExceptionFactory
                    .getInstance()
                    .registerNewAwesomeException(
                            TAG,
                            ExceptionCode.CODE_INSUFFICIENT_PERMISSIONS,
                            "SQLitePrimitivesDB is not available",
                            ExceptionCode.DETAILED_INSUFFICIENT_PERMISSIONS+".setLocalization");
            return false;
        }
    }

    public String getLocalization(Context context){
        String languageCode = null;

        try {
            Locale currentLocale = context.getResources().getConfiguration().locale;
            languageCode = currentLocale.toLanguageTag().toLowerCase();
            languageCode = SQLitePrimitivesDB
                    .getInstance(context)
                    .getString(
                            context,
                            "localization",
                            "languageCode",
                            languageCode
                    );

        } catch (Exception e) {
            ExceptionFactory
                    .getInstance()
                    .registerNewAwesomeException(
                            TAG,
                            ExceptionCode.CODE_INSUFFICIENT_PERMISSIONS,
                            "SQLitePrimitivesDB is not available",
                            ExceptionCode.DETAILED_INSUFFICIENT_PERMISSIONS+".getLocalization");
            return languageCode;
        }

        return languageCode;
    }
}

File: ios/AwnCore/Classes/managers/LocalizationManager.swift
//
//  LocalizationManager.swift
//  IosAwnCore
//
//  Created by Rafael Setragni on 18/02/23.
//

import Foundation

class LocalizationManager {
    static let shared = LocalizationManager()
    
    func setLocalization(languageCode: String? = nil) -> Bool {
        let langCode =
            languageCode ??
            Locale.preferredLanguages[0]
        
        switch SQLitePrimitivesDB.shared.setString(
            tag: "localization",
            key: "languageCode",
            value: langCode
                .lowercased()
                .replacingOccurrences(of: "_", with: "-")
        ) {
        case .success():
            return true
        case .failure(let error):
            let awnError = ExceptionFactory.shared.createNewAwesomeException(
                className: "LocalizationManager",
                code: ExceptionCode.CODE_INSUFFICIENT_PERMISSIONS,
                message: "SQLitePrimitivesDB is not available \(error.localizedDescription)",
                detailedCode: ExceptionCode.DETAILED_INSUFFICIENT_PERMISSIONS+".setLocalization"
            )
            print(awnError)
            return false
        }
    }
    
    func getLocalization() -> String {
        let appLangCode = Locale.preferredLanguages[0]
        
        switch SQLitePrimitivesDB.shared.getString(
            tag: "localization",
            key: "languageCode"
        ) {
            
        case .success(let value):
            return (value ?? appLangCode)
                .lowercased()
                .replacingOccurrences(of: "_", with: "-")
            
        case .failure(let error):
            let error = ExceptionFactory.shared.createNewAwesomeException(
                className: "LocalizationManager",
                code: ExceptionCode.CODE_INSUFFICIENT_PERMISSIONS,
                message: "SQLitePrimitivesDB is not available \(error.localizedDescription)",
                detailedCode: ExceptionCode.DETAILED_INSUFFICIENT_PERMISSIONS+".getLocalization"
            )
            print(error)
            return appLangCode
                .lowercased()
                .replacingOccurrences(of: "_", with: "-")
        }
    }
}
